gridstatus/ercot.py

```diff
@@ -576,6 +579,120 @@ def get_spp(
             )
         return Ercot._finalize_spp_df(df, settlement_point_field, locations)
 
+    def get_as_prices(
```

Comment:
[adding for context]

Basic example: When you call the function with the `@support_date_range("1Y")` and `date=1/1/2020, end=1/1/2023`, it runs a loop over the following timestamps:
```
dates=[
    Timestamp('2020-01-01 00:00:00-0600', tz='US/Central'), 
    Timestamp('2020-12-31 00:00:00-0600', tz='US/Central', freq='A-DEC'), 
    Timestamp('2021-12-31 00:00:00-0600', tz='US/Central', freq='A-DEC'), 
    Timestamp('2022-12-31 00:00:00-0600', tz='US/Central', freq='A-DEC'), 
    Timestamp('2023-01-01 00:00:00-0600', tz='US/Central')
]
```
so you actually do get multiple distinct year values per iteration passed to the function (per your comment). There's not more than 1 year between dates, but that doesn't necessarily matter because the file identifier is a distinct year integer.

The function actually does still work if you use the decorator, but you still need the logic checking if there are different discreet years between `start` and `end`.